### PR TITLE
Add granular protobuf errors for ActiveRemote

### PR DESCRIPTION
**Pull request by ryanbjones**
From https://github.com/liveh2o/active_remote/issues/39

Example set of errors https://github.com/ruby-protobuf/protobuf/blob/master/lib/protobuf/rpc/error/client_error.rb

This will make it so we bubble up the type of error given from Protobuf

@liveh2o @abrandoned @mmmries

### CODE REVIEW

File: lib/active_remote/rpc_adapters/protobuf_adapter.rb [hunk 1]
```diff
@@ -36,6 +37,33 @@ def execute(rpc_method, request_args)
...
+      def protobuf_error_class(error)
+        case error.error_type
+        when ::Protobuf::Socketrpc::ErrorReason::BAD_REQUEST_DATA
+          ::ActiveRemote::BadRequestDataError
+        when ::Protobuf::Socketrpc::ErrorReason::BAD_REQUEST_PROTO
+          ::ActiveRemote::BadRequestProtoError
+        when ::Protobuf::Socketrpc::ErrorReason::SERVICE_NOT_FOUND
+          ::ActiveRemote::ServiceNotFoundError
+        when ::Protobuf::Socketrpc::ErrorReason::METHOD_NOT_FOUND
+          ::ActiveRemote::MethodNotFound
```
Comment (jgaxn) [inline lib/active_remote/rpc_adapters/protobuf_adapter.rb]: should be `MethodNotFoundError`

Comment (ryanbjones) [inline lib/active_remote/rpc_adapters/protobuf_adapter.rb]: 👍  nice catch!

**Comment by jgaxn** [comment]
LGTM :shipit:

**Comment by mmmries** [comment]
:shipit: nice call on making all of the granular error inherit from `ActiveRemoteError` so all the existing `catch` clauses that people are using will still work.

**Comment by liveh2o** [comment]
Can you provide some additional context as to what problem this is solving? I mainly want to understand the gaps better.

**Pull request merged**